test/tests/performance/pbs_preemptperformance.py

```diff
@@ -513,24 +513,25 @@ def test_preemption_with_soft_limits(self):
         self.server.manager(MGR_CMD_CREATE, QUEUE, a, qname)
 
         self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
-        a = {ATTR_l + '.select=1:ncpus': 1, ATTR_q: 'highp'}
-        jid = ""
+        a = {ATTR_l + '.select=1:ncpus': 1, ATTR_q: qname}
         for _ in range(2000):
             j = Job(TEST_USER2, attrs=a)
             j.set_sleep_time(3000)
-            jid = self.server.submit(j)
+            ljid = self.server.submit(j)
         scycle = time.time()
         self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
-        (_, str1) = self.scheduler.log_match("Starting Scheduling Cycle",
-                                             n='ALL', starttime=int(scycle),
+
+        # Rollback 2000 job ids to get to the first high priority job id
+        fjid = str((int(ljid.split('.')[0]) - 2000)) + '.' + ljid.split('.')[1]
```

Comment:
Don't you submit the high priority job separately?  Why do you need to figure out which job is the high priority job?

Reply:
I submit 2000 high priority jobs in a loop, I have the last jobid in ljid variable, from that job id I compute which is the first high priority job id. Once I have both the job ids, I run log_match() for both jobs to compute the time it takes to run 2000 high priority job by preempting 2000 normal jobs.

Reply:
might be more readable if you set fjid in the loop:
"if fjid is None:
    fjid = self.server.submit(j)
else:
    ljid = self.server.submit(j)"

Reply:
@agrawalravi90 I've addressed the code for readability like the way you suggested, Please have a look at the latest commit.